### ISSUE TITLE
Question: Async/Await inside of Subscribe

### ISSUE THREAD

**Issue by Calidus** [danielwertheim/mynatsclient#46]
Does MyNatsClient support using async/await inside of a subscribe method? The code below will compile but I think its just firing and forgetting tasks. The use case I was thinking of was I want subscribe to a queue than write then write message to a database or file.

`await _natsClient.SubAsync("foo", stream => stream.Subscribe(async msg => await someMethodAsync(msg)));`

**Comment by bergi9** [comment]
TL:DR; No.

The argument of `stream.Subscribe(..)` accepts an `Action<T>`.
See: [NatsObservableExtensions.cs#L24](https://github.com/danielwertheim/mynatsclient/blob/36df86dd366eb66fc95f095aa9905c3a6dcf8388/src/MyNatsClient/Rx/NatsObservableExtensions.cs#L24) and [NatsObserver.cs#L9](https://github.com/danielwertheim/mynatsclient/blob/36df86dd366eb66fc95f095aa9905c3a6dcf8388/src/MyNatsClient/NatsObserver.cs#L9)
So it's not awaitable.
You can use the `Task.Wait()` like `stream.Subscribe(msg => someMethodAsync(msg).Wait())` if you want to process the messages sequentially inside of the subscription and also block NyNatsClient's internal message processing.
Or use `.ObserveOn(...)` with an `IScheduler` implementation to fulfill your requirement. For example running the observer in an other single thread, taskpool or eventloop. [scheduling with rx](http://introtorx.com/Content/v1.0.10621.0/15_SchedulingAndThreading.html).
